display: stm32: ltdc: minor fixes

ajarmouni-st opened the pull request:
Renames the `display-controller` property of `st,stm32-ltdc` to `panel-controller`.

Clarifies why NULL checks are necessary for panel controller device inside blanking_on/off functions in LTDC driver,
& fixes return value when the panel controller's phandle is not passed to LTDC in devicetree. 

Addresses https://github.com/zephyrproject-rtos/zephyr/pull/68105#discussion_r1476504953

decsny reviewed (dismissed):
will need to update a migration guide to change the DT binding property name

kartben reviewed (dismissed):
I would squash the commit updating the migration guide into the first commit as there is no reason to keep it separate IMO, but I will make this non-blocking (and doing the update shouldn't dismiss existing reviews anyway)

mathieuchopstm commented on drivers/display/display_stm32_ltdc.c:
```diff
@@ -267,13 +263,9 @@ static int stm32_ltdc_display_blanking_off(const struct device *dev)
-
-	if (!device_is_ready(config->display_controller)) {
+	if (!device_is_ready(display_dev)) {
 		LOG_ERR("Display device %s not ready", display_dev->name);
```
Can't use `display_dev->name` (`display_dev` may be NULL).

Also, nit:
```suggestion
		LOG_ERR("Panel device %s not ready", display_dev->name);
```

GeorgeCGV replied:
The `device_is_ready` already checks for `null`:
```
	if (dev == NULL) {
		return false;
	}
```

mathieuchopstm replied:
If `display_dev` is NULL, then `device_is_ready(display_dev)` returns false, and the `if` block is executed.

mathieuchopstm commented on drivers/display/display_stm32_ltdc.c:
```diff
@@ -250,11 +250,7 @@ static int stm32_ltdc_read(const struct device *dev, const uint16_t x,
+	const struct device *display_dev = config->panel_controller;
 
 	if (!device_is_ready(display_dev)) {
 		LOG_ERR("Display device %s not ready", display_dev->name);
```
Same remark.

Same nit:
```suggestion
		LOG_ERR("Panel device %s not ready", display_dev->name);
```

mathieuchopstm reviewed (changes requested):
LGTM besides the following:

ajarmouni-st committed:
binding: stm32-ltdc: change display-controller to panel-controller

display-controller is ambiguous. change it to panel-controller.
Update LTDC driver & migration guide.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
drivers: display: stm32_ltdc: fix return value

Blanking On/Off calls should not return 0 when there is no panel
controller to forward them to, instead they should return ENOSYS to signal
to the application that they were not actually executed.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

GeorgeCGV commented:
@mathieuchopstm, @ajarmouni-st,

The `device_is_ready` already checks for `null`:
https://github.com/zephyrproject-rtos/zephyr/blob/9b9d455f0231f52ea75e7030c98d331937d05f47/kernel/device.c#L142-L153

No need to add extra device `null` checks like `if (panel_dev == NULL) ...`

ajarmouni-st commented:
> The `device_is_ready` already checks for `null`:
> 
> No need to add extra device `null` checks like `if (panel_dev == NULL) ...`

@GeorgeCGV The idea here is to isolate the `dev == NULL` case from `dev->state->initialized == false` case.

GeorgeCGV commented:
@ajarmouni-st if the goal is to allow people to control blanking manually in the app, then the `stm32_ltdc_init` could print an info about `panel_controller` being `null` and blanking calls could be way smaller with some macro magic (_as that is user's intention_). That could reduce the footprint a bit.

If the goal is to not allow that, then it would be better to fil early by replacing `DEVICE_DT_GET_OR_NULL` with `DEVICE_DT_GET`.
Then the `null` checks are irrelevant.

But ok.

ajarmouni-st commented:
> it would be better to fail early by replacing `DEVICE_DT_GET_OR_NULL` with `DEVICE_DT_GET`

That would not be desirable because we don't know if the app will actually call a display API function that needs to be forwarded to the panel controller.

ajarmouni-st commented:
> blanking calls could be way smaller with some macro magic
 I am not sure I understand what you mean by this.

GeorgeCGV commented:
> > it would be better to fail early by replacing `DEVICE_DT_GET_OR_NULL` with `DEVICE_DT_GET`
> 
> That would not be desirable because we don't know if the app will actually call a display API function that needs to be forwarded to the panel controller.

Some drivers call blanking on upon init, some don't. Same goes for blanking off.

> > blanking calls could be way smaller with some macro magic
> 
> @GeorgeCGV I am not sure I understand what you mean by this.

If LTDC's blanking control is intended to be optional, then it is possible to define blanking differently depending on the `panel_controller` value. That allows to save some bytes when `panel_controller` is not provided. But that is more into "optimization".

ajarmouni-st commented:
@GeorgeCGV I don't know if this is relevant or not, but `panel-controller` is useful to redirect in a transparent way the call of any display API function that cannot be performed by LTDC itself but it is implemented by the panel controller driver.

ajarmouni-st commented:
@decsny Please take another look, thanks!

ajarmouni-st commented:
@danieldegrasse could you help review this, thanks!

jfischer-no reviewed (changes requested):
```diff
-	display-controller = <&ili9341>;
+	panel-controller = <&ili9341>;
```
Whether display-controller is ambiguous or not, it is not a bug.

jfischer-no commented:
@ajarmouni-st But `drivers: display: stm32_ltdc: fix return value ` is still valid, why not just drop the first commit?
